# Cardiac Patient Monitoring System

## Project Description
In this project, it is intended to use the machine learning techniques to work on the health information related to heart disease. A machine learning pipeline would be created to analyze the classifiers using heart disease health indicators.
**Heart Disease Health Indicators dataset** is chosen for the current project.

## Environment Setup
The project uses Python and Jupyter Notebook.
### Import Libraries

In [1]:
import numpy as np
import pandas as pd

## Choice of Dataset
- Data source: Kaggle
https://www.kaggle.com/datasets/alexteboul/heart-disease-health-indicators-dataset
- Dataset name: Heart Disease Health Indicators Dataset
- Description: The dataset includes health, lifestyle, and demographic data about heart disease.
- Target Variable:
HeartDisease
- Target class values:
Yes – there is heart disease
No – there is no heart disease
### Why this Dataset?
- The dataset was chosen for the following reasons:
- It is clear what kind of classification we should do.
- It includes both numeric and categorical variables.
- There are enough observations for machine learning.
- It is appropriate for exploratory analysis and classification.
- It is good for both supervised and unsupervised learning that will be done later in the project.

## Load the Dataset

In [4]:
df = pd.read_csv("heart_disease_health_indicators_BRFSS2015.csv")

## Initial Dataset Inspection (First Ten Rows,Dataset Shape,Column Names,Data Types and Non-Null Values)

In [5]:
df.head(10)

,HeartDiseaseorAttack,HighBP,HighChol,CholCheck,BMI,Smoker,Stroke,Diabetes,PhysActivity,Fruits,...,AnyHealthcare,NoDocbcCost,GenHlth,MentHlth,PhysHlth,DiffWalk,Sex,Age,Education,Income
0,0.0,1.0,1.0,1.0,40.0,1.0,0.0,0.0,0.0,0.0,...,1.0,0.0,5.0,18.0,15.0,1.0,0.0,9.0,4.0,3.0
1,0.0,0.0,0.0,0.0,25.0,1.0,0.0,0.0,1.0,0.0,...,0.0,1.0,3.0,0.0,0.0,0.0,0.0,7.0,6.0,1.0
2,0.0,1.0,1.0,1.0,28.0,0.0,0.0,0.0,0.0,1.0,...,1.0,1.0,5.0,30.0,30.0,1.0,0.0,9.0,4.0,8.0
3,0.0,1.0,0.0,1.0,27.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,0.0,0.0,0.0,0.0,11.0,3.0,6.0
4,0.0,1.0,1.0,1.0,24.0,0.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,3.0,0.0,0.0,0.0,11.0,5.0,4.0
5,0.0,1.0,1.0,1.0,25.0,1.0,0.0,0.0,1.0,1.0,...,1.0,0.0,2.0,0.0,2.0,0.0,1.0,10.0,6.0,8.0
6,0.0,1.0,0.0,1.0,30.0,1.0,0.0,0.0,0.0,0.0,...,1.0,0.0,3.0,0.0,14.0,0.0,0.0,9.0,6.0,7.0
7,0.0,1.0,1.0,1.0,25.0,1.0,0.0,0.0,1.0,0.0,...,1.0,0.0,3.0,0.0,0.0,1.0,0.0,11.0,4.0,4.0
8,1.0,1.0,1.0,1.0,30.0,1.0,0.0,2.0,0.0,1.0,...,1.0,0.0,5.0,30.0,30.0,1.0,0.0,9.0,5.0,1.0
9,0.0,0.0,0.0,1.0,24.0,0.0,0.0,0.0,0.0,0.0,...,1.0,0.0,2.0,0.0,0.0,0.0,1.0,8.0,4.0,3.0


In [22]:
df.shape

(229781, 22)

In [23]:
df.columns

Index(['HeartDiseaseorAttack', 'HighBP', 'HighChol', 'CholCheck', 'BMI',
       'Smoker', 'Stroke', 'Diabetes', 'PhysActivity', 'Fruits', 'Veggies',
       'HvyAlcoholConsump', 'AnyHealthcare', 'NoDocbcCost', 'GenHlth',
       'MentHlth', 'PhysHlth', 'DiffWalk', 'Sex', 'Age', 'Education',
       'Income'],
      dtype='str')

In [24]:
df.info()

<class 'pandas.DataFrame'>
Index: 229781 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   HeartDiseaseorAttack  229781 non-null  float64
 1   HighBP                229781 non-null  float64
 2   HighChol              229781 non-null  float64
 3   CholCheck             229781 non-null  float64
 4   BMI                   229781 non-null  float64
 5   Smoker                229781 non-null  float64
 6   Stroke                229781 non-null  float64
 7   Diabetes              229781 non-null  float64
 8   PhysActivity          229781 non-null  float64
 9   Fruits                229781 non-null  float64
 10  Veggies               229781 non-null  float64
 11  HvyAlcoholConsump     229781 non-null  float64
 12  AnyHealthcare         229781 non-null  float64
 13  NoDocbcCost           229781 non-null  float64
 14  GenHlth               229781 non-null  float64
 15  MentHlth        

These checks provide basic insights into the structure of the dataset, number of observations, columns, and types of data.

## Data Dictionary
The description and data type were based on the data documentation provided
with the dataset. The dataset was generated using the Behavioral Risk Factor
Surveillance System  (BRFSS) of the CDC.

| Column | Description | Type |
|---|---|---|
| `HeartDiseaseorAttack` | Whether the person has ever reported heart disease or a heart attack | Binary Categorical |
| `HighBP` | Whether the person has high blood pressure | Binary Categorical |
| `HighChol` | Whether the person has high cholesterol | Binary Categorical |
| `CholCheck` | Whether cholesterol has been checked in the last 5 years | Binary Categorical |
| `BMI` | Body Mass Index | Numerical |
| `Smoker` | Whether the person has smoked at least 100 cigarettes in their lifetime | Binary Categorical |
| `Stroke` | Whether the person has ever had a stroke | Binary Categorical |
| `Diabetes` | Diabetes status | Categorical |
| `PhysActivity` | Whether the person has participated in physical activity in the past 30 days | Binary Categorical |
| `Fruits` | Whether the person consumes fruit at least once per day | Binary Categorical |
| `Veggies` | Whether the person consumes vegetables at least once per day | Binary Categorical |
| `HvyAlcoholConsump` | Whether the person is a heavy alcohol consumer | Binary Categorical |
| `AnyHealthcare` | Whether the person has any kind of health care coverage | Binary Categorical |
| `NoDocbcCost` | Whether there was a time in the past 12 months when the person needed a doctor but could not afford it | Binary Categorical |
| `GenHlth` | General health status reported by the person | Ordinal Categorical |
| `MentHlth` | Number of days mental health was not good during the past 30 days | Numerical |
| `PhysHlth` | Number of days physical health was not good during the past 30 days | Numerical |
| `DiffWalk` | Whether the person has serious difficulty walking or climbing stairs | Binary Categorical |
| `Sex` | Sex of the Patient | Binary Categorical |
| `Age` | Age category of the Patient represented by an encoded value | Ordinal Categorical |
| `Education` | Education level represented by an encoded value | Ordinal Categorical |
| `Income` | Income category represented by an encoded value | Ordinal Categorical |

## Initial Data Cleaning

### Check Missing Values
This checks whether any columns contain Null values.

In [25]:
df.isnull().sum()

HeartDiseaseorAttack    0
HighBP                  0
HighChol                0
CholCheck               0
BMI                     0
Smoker                  0
Stroke                  0
Diabetes                0
PhysActivity            0
Fruits                  0
Veggies                 0
HvyAlcoholConsump       0
AnyHealthcare           0
NoDocbcCost             0
GenHlth                 0
MentHlth                0
PhysHlth                0
DiffWalk                0
Sex                     0
Age                     0
Education               0
Income                  0
dtype: int64

### Check Duplicate Rows
If duplicate rows exist, they can be removed

In [26]:
df.duplicated().sum()

np.int64(0)

### Check Unique Values
This provides an initial check for unexpected or inconsistent values for example the column HeartDiseaseorAttack should have only two variables.

In [27]:
for column in df.columns:
    print(column, df[column].unique())

HeartDiseaseorAttack [0. 1.]
HighBP [1. 0.]
HighChol [1. 0.]
CholCheck [1. 0.]
BMI [40. 25. 28. 27. 24. 30. 34. 26. 33. 21. 23. 22. 38. 32. 37. 31. 29. 20.
 35. 45. 39. 19. 47. 18. 36. 43. 55. 49. 42. 17. 16. 41. 44. 50. 59. 48.
 52. 46. 54. 57. 53. 14. 15. 51. 58. 63. 61. 56. 74. 62. 64. 66. 73. 85.
 60. 67. 65. 70. 82. 79. 92. 68. 72. 88. 96. 13. 81. 71. 75. 12. 77. 69.
 76. 87. 89. 84. 95. 98. 91. 86. 83. 80. 90. 78.]
Smoker [1. 0.]
Stroke [0. 1.]
Diabetes [0. 2. 1.]
PhysActivity [0. 1.]
Fruits [0. 1.]
Veggies [1. 0.]
HvyAlcoholConsump [0. 1.]
AnyHealthcare [1. 0.]
NoDocbcCost [0. 1.]
GenHlth [5. 3. 2. 4. 1.]
MentHlth [18.  0. 30.  3.  5. 15. 10.  6. 20.  2. 25.  1.  4.  7.  8. 21. 14. 26.
 29. 16. 28. 11. 12. 24. 17. 13. 27. 19. 22.  9. 23.]
PhysHlth [15.  0. 30.  2. 14. 28.  7. 20.  3. 10.  1.  5. 17.  4. 19.  6. 12. 25.
 27. 21. 22.  8. 29. 24.  9. 16. 18. 23. 13. 26. 11.]
DiffWalk [1. 0.]
Sex [0. 1.]
Age [ 9.  7. 11. 10.  8. 13.  4.  6.  2. 12.  5.  1.  3.]
Education [4. 6. 3. 5

## Target Variable
The target variable is:
**HeartDiseaseorAttack**
It represents whether a patient  having heart disease.
This makes the machine-learning problem a binary classification problem.

In [28]:
df["HeartDiseaseorAttack"].value_counts()

HeartDiseaseorAttack
0.0    206064
1.0     23717
Name: count, dtype: int64

The target variable HeartDiseaseorAttack includes two categories:
- 0.0: 206,064 cases
- 1.0: 23,717 cases
There are no missing values in the target variable and two valid categories; therefore, it is a binary classification target.

## First Observations on the Quality of the Data
- The number of rows is 253,680, and there are 22 columns.
- There are no missing values in the data.
- There are no duplicates in the rows.
- All the columns have the type float64.
- Nevertheless, not all the variables are numeric. Some variables are categorical, and they have numerical encoding.
- BMI, MentHlth, and PhysHlth are numeric variables.
- Age, Education, and Income are ordinal categorical variables coded numerically.
- The target variable HeartDiseaseorAttack is binary, and it takes the values 0 and 1.